### PR TITLE
Move fleet functionality, test cases, minor game.js fix

### PR DESCRIPTION
**Pull request by kramerje**
Implemented move fleet functionality, added test cases, and made sure the console wouldn't throw an error after user places all their ships (around line 300 in game.js, the el tries to access undefined after all ships are placed).

### CODE REVIEW

File: src/main/java/assets/game.js [hunk 1]
```diff
@@ -320,6 +341,28 @@ function sonar_call_back(data){
...
 }
+
+
+function move_fleet_call_back(data){
+// reset isMoveFleet variable to false again (so in normal attack mode)
+    isMoveFleet = false;
+    game = data;
+    redrawGrid();
+
+    move_fleet_uses++;
```
Comment (aalokb) [inline src/main/java/assets/game.js]: Good that you allocated new squares to handle the copying of the occupied squares, as any other method would have screwed up the consistency of the square placement...

File: src/main/java/assets/game.js [hunk 2]
```diff
@@ -320,6 +341,28 @@ function sonar_call_back(data){
...
+    if(move_fleet_uses == 2){
+        $("#move_fleet_box").hide();
+    }
+    else{
+        //restore the button back to popped up state
+        $("#up_image").css("opacity", 1);
```
Comment (aalokb) [inline src/main/java/assets/game.js]: I see you used the same logic for the button push down, as the one used in sonar attack, this should work, but be careful about a user possibly repeatedly clicking on the button and screwing up the logic.

File: src/main/java/assets/game.js [hunk 3]
```diff
@@ -692,6 +744,96 @@ function sonar_attack(){
...
+
+    if(move_fleet_clicked_down){
+        isMoveFleet = true;
+    }
+    else{
+        isMoveFleet = false;
+    }
+}
+
+function move_fleet_down(){
```
Comment (aalokb) [inline src/main/java/assets/game.js]: Awesome, very solid functionality, and I'm glad you modularized it into four seperate functions, great job. Just beautiful. I'm tearing up right now just looking at this. I think I need a moment....jkhjgrdy6uf7897gt8oph0o;lnph;
That was me wiping the tears off of my keyboard.

Comment (kramerje) [reply src/main/java/assets/game.js]: Strong tears you got there friend

**Pull request merged**